examples/todo-list/flutter-todo-list/lib/screens/todo.dart

```diff
@@ -0,0 +1,300 @@
+    _taskEditingController.dispose();
+  }
+
+  Future<void> displayTextInputDialog({
+    required BuildContext context,
+    void Function()? onTab,
+    required TextEditingController taskCtrl,
+  }) async {
+    return showDialog(
+      context: context,
+      builder: (context) {
+        return AlertDialog(
+          title: const Text('Add a todo'),
+          content: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              TextField(
+                controller: taskCtrl,
+                decoration:
+                    const InputDecoration(hintText: 'Add title of todo'),
+              ),
+              smallGap,
+              MaterialButton(
+                onPressed: onTab,
+                child: const Text(
+                  'Add Task',
+                  style: TextStyle(
+                    color: Colors.white,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+                color: const Color(0xff33b27b),
+              )
+            ],
+          ),
+        );
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Colors.black,
+        automaticallyImplyLeading: false,
+        title: Text('Supabase Todo'),
+        actions: <Widget>[
+          IconButton(
+            icon: const Icon(Icons.exit_to_app),
+            onPressed: () async {
+              await logOut(context);
+              Navigator.pop(context, '/');
+            },
+          ),
+        ],
+      ),
+      body: loading
+          ? const Center(child: CircularProgressIndicator())
+          : ConstrainedBox(
+              constraints: const BoxConstraints(),
+              child: todos.isEmpty
+                  ? const Center(
+                      child: Text('No Task, add new task'),
+                    )
+                  : ListView.builder(
+                      itemCount: todos.length,
+                      itemBuilder: (context, index) {
+                        return ListTile(
+                          //Long Press to edit
+                          onLongPress: () {
+                            displayTextInputDialog(
+                              context: context,
+                              taskCtrl: _taskEditingController,
+                              onTab: () async {
+                                await editTask(
+                                  id: todos[index]['id'].toString(),
+                                  task: _taskEditingController.text,
+                                );
+                                Navigator.pop(context);
+                                setState(() {
+                                  loading = true;
+                                });
+                                List<dynamic> value = await getTodo();
+                                todos.clear();
+                                setState(() {});
```

Comment:
I think we can get rid of this once since setState is being called few lines down!
```suggestion
```